zerver/tests/test_user_groups.py

```diff
@@ -350,42 +351,98 @@ def test_update_members_of_user_group(self) -> None:
         self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 1)
 
         othello = self.example_user("othello")
-        add = [othello.id]
-        params = {"add": orjson.dumps(add).decode()}
-        result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
-        self.assert_json_success(result)
+        params = {"add": orjson.dumps([othello.id]).decode()}
+
+        events: List[Mapping[str, Any]] = []
+
+        with self.tornado_redirected_to_list(events, expected_num_events=2):
+            result = self.client_post(f"/json/user_groups/{user_group.id}/members", info=params)
+            self.assert_json_success(result)
+
         self.assertEqual(UserGroupMembership.objects.filter(user_group=user_group).count(), 2)
         members = get_direct_memberships_of_users(user_group, [hamlet, othello])
         self.assert_length(members, 2)
 
+        # A notification message is sent for adding to user group
+        msg_event = [e for e in events if e["event"]["type"] == "message"]
+        self.assert_length(msg_event, 1)
+        expected_notification = (
+            f"{silent_mention_syntax_for_user(hamlet)} subscribed you to the group @_*support*."
+        )
+        self.assertEqual(msg_event[0]["event"]["message_dict"]["content"], expected_notification)
```

Comment:
Now reading this again, instead of testing the event, we should check the actual `Message` object and its content. You can either use `get_last_message` helper or can create something like `get_topic_messages` that would be used to get PMs with a particular user.